### PR TITLE
Include GoMock and go-junit-report versions

### PR DESCRIPTION
**Pull request by jasonodoom**
#  #

## 🗣 Description ##

This PR is to track the versions for [GoMock](https://github.com/golang/mock) and [go-junit-report](https://github.com/jstemmer/go-junit-report) into this repo so it can be consumed/used in GitHub Actions workflows. 

## 💭 Motivation and context ##

This change is based on the discussion https://github.com/cisagov/skeleton-golang-package/pull/2#discussion_r1177074251

This will allow us to track and store version numbers for these dependencies in one location as we've been doing.

## 🧪 Testing ##

Pre-commit verified syntactical accuracy. 

**Edit:**
I (@mcdonnnj) created a [test branch in  `cisagov/skeleton-golang-package`](https://github.com/cisagov/skeleton-golang-package/tree/testing/setup-env) to verify the functionality of this action. I verified that the two project versions added here were usable to install the projects in GitHub Actions.

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] All relevant type-of-change labels have been added.
- [X] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [X] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [X] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

File: action.yml [hunk 1]
```diff
@@ -33,3 +39,9 @@ runs:
     - id: terraform-docs
       run: echo "version=v0.16.0" >> $GITHUB_OUTPUT
       shell: bash
+    - id: go-mock-version
+      run: echo "version=1.6.0>> $GITHUB_OUTPUT
+      shell: bash
+    - id: go-junit-report-version
+      run: echo "version=2.0.0" >> $GITHUB_OUTPUT
+      shell: bash
```
Comment (mcdonnnj) [inline action.yml]: Please put these in alphabetical order with the other steps and remove the `-version` from these `id`s.

File: action.yml [hunk 2]
```diff
@@ -18,6 +18,12 @@ outputs:
   terraform-docs-version:
     description: The version of terraform-docs to use.
     value: ${{ steps.terraform-docs.outputs.version }}
+  go-mock-version:
+    description: The version of go mock to use.
+    value: ${{ steps.go-mock-version.outputs.version }}
+  go-junit-report-version:
+    description: The version of go junit report to use.
+    value: ${{ steps.go-junit-report.outputs.version }}
```
Comment (mcdonnnj) [inline action.yml]: Please reorder these to match the order in the steps below this in the file.

```suggestion
  go-mock-version:
    description: The version of go mock to use.
    value: ${{ steps.go-mock.outputs.version }}
  go-junit-report-version:
    description: The version of go junit report to use.
    value: ${{ steps.go-junit-report.outputs.version }}
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: action.yml (hunk 2 above)
Comment (jasonodoom) [reply action.yml]: Sorry I saw your suggestion after my commit. Please see commit ecbc0939d1f16bce6d60cf088943fd6904f530dc

File: action.yml (hunk 1 above)
Comment (jasonodoom) [reply action.yml]: Please see commit ecbc0939d1f16bce6d60cf088943fd6904f530dc

File: action.yml [hunk 3]
```diff
@@ -6,6 +6,12 @@ branding:
   color: purple
 description: Setup a shared GitHub Actions workflow environment.
 outputs:
+  go-junit-report-version:
+    description: The version of go junit report to use.
+    value: ${{ steps.go-junit-report.outputs.version }}
+  go-mock-version:
+    description: The version of go mock to use.
```
Comment (dav3r) [inline action.yml]: Clarify which repos are being versioned:
```suggestion
    description: The version of jstemmer/go-junit-report to use.
    value: ${{ steps.go-junit-report.outputs.version }}
  go-mock-version:
    description: The version of golang/mock to use.
```

**Review by dav3r — CHANGES_REQUESTED** [review]
In addition to the one suggestion I noted, please also do the following:
* Include links to https://github.com/golang/mock and https://github.com/jstemmer/go-junit-report in the PR description.
* Update the README to include `go-junit-report-version` and `go-mock-version` in the [Outputs table](https://github.com/cisagov/setup-env-github-action#outputs).

File: README.md [hunk 4]
```diff
@@ -13,11 +13,14 @@ None
...
 | Name              | Description |
 |-------------------|-------------|
+| go-junit-report-version | The version of [go-junit-report](https://github.com/jstemmer/go-junit-report) to use. |
+| go-mock-version | The version of [go-mock](https://github.com/golang/mock) to use. |
 | packer-version | The version of [Packer](https://packer.io) to use. |
 | shfmt-version | The version of [shfmt](https://github.com/mvdan/sh#shfmt) to use. |
 | terraform-version | The version of [Terraform](https://terraform.io) to use. |
 | terraform-docs-version | The version of [terraform-docs](https://github.com/terraform-docs/terraform-docs) to use. |
 
+
```
Comment (jsf9k) [inline README.md]: Remove unnecessary blank line.
```suggestion
```

File: action.yml [hunk 5]
```diff
@@ -18,9 +24,16 @@ outputs:
...
     value: ${{ steps.terraform-docs.outputs.version }}
+
 runs:
   using: "composite"
   steps:
+    - id: go-junit-report
+      run: echo "version=2.0.0" >> $GITHUB_OUTPUT
+      shell: bash
+    - id: go-mock
+      run: echo "version=1.6.0>> $GITHUB_OUTPUT
```
Comment (jsf9k) [inline action.yml]: Add a space for consistency.
```suggestion
      run: echo "version=1.6.0 >> $GITHUB_OUTPUT
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
I am requesting a few very minor changes.

**Review by jsf9k — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
👍 👍

File: README.md [hunk 6]
```diff
@@ -13,6 +13,8 @@ None
 
 | Name              | Description |
 |-------------------|-------------|
+| go-junit-report-version | The version of [go-junit-report](https://github.com/jstemmer/go-junit-report) to use. |
+| go-mock-version | The version of [go-mock](https://github.com/golang/mock) to use. |
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
| gomock-version | The version of [GoMock](https://github.com/golang/mock) to use. |
```

File: action.yml [hunk 7]
```diff
@@ -18,9 +24,16 @@ outputs:
...
     description: The version of terraform-docs to use.
     value: ${{ steps.terraform-docs.outputs.version }}
+
 runs:
   using: "composite"
   steps:
+    - id: go-junit-report
+      run: echo "version=2.0.0" >> $GITHUB_OUTPUT
+      shell: bash
+    - id: go-mock
```
Comment (mcdonnnj) [inline action.yml]:
```suggestion
    - id: gomock
```

File: action.yml [hunk 8]
```diff
@@ -6,6 +6,12 @@ branding:
   color: purple
 description: Setup a shared GitHub Actions workflow environment.
 outputs:
+  go-junit-report-version:
+    description: The version of jstemmer/go-junit-report to use.
+    value: ${{ steps.go-junit-report.outputs.version }}
+  go-mock-version:
+    description: The version of golang/mock to use.
+    value: ${{ steps.go-mock.outputs.version }}
```
Comment (mcdonnnj) [inline action.yml]:
```suggestion
  gomock-version:
    description: The version of golang/mock to use.
    value: ${{ steps.gomock.outputs.version }}
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Updating to reflect how the GoMock project self-references.

**Review by mcdonnnj — APPROVED** [review]

**Comment by jsf9k** [comment]
@mcdonnnj, is there any reason this PR can't be merged now?

**Review by dav3r — APPROVED** [review]
👍 👍

**Review by jsf9k — APPROVED** [review]

**Pull request merged**